numjuggler/likefunc.py

```diff
@@ -59,7 +59,7 @@ def __init__(self, log=False):
         self.ld = {}
 
         # Flag to store log
-        self.log = log
+        self.__lf = log
```

Comment:
In the original code, i.e. ``self.log = log`` the setter is called that sets the particular variant of the ``__call__`` method. In your case the ``__call2`` method will not be used when ``True`` is passed directly to the constructor.

Reply:
Somewhere in the code there was a log call  to the attribute __lf, which appears to be bool value at the call moment. I've decide that this simple misprint and fixed this. Probably my fix doesn't cover __call2 functionality, but allowed to run renumbering.
We have to add test for logical branch where__call2 should be invoked.

Reply:
Thanks, indeed, there is the bug related to the ``log`` property`. I'll fix this directly on the master branch.

Reply:
self.log=... creates attribute log
@property
def log()...         - creates method
both in the same class,  don't you think this is bad? These two guys must have different names to me.

Reply:
When ``def log(self)`` is [decorated](https://docs.python.org/2/library/functions.html#property) with ``@log.setter``, setting the respective attribute, i.e. ``self.log = value`` actually calls ``self.log(value)``.

Reply:
Looks like all this is fixed in the next pull request